File: frontend/src/types/analysis.ts
export interface MACDPoint {
  macd: number | null
  signal: number | null
  histogram: number | null
}

export interface BollingerPoint {
  upper: number | null
  middle: number | null
  lower: number | null
}

export interface IndicatorPoint {
  timestamp: string
  close: number
  rsi: number | null
  macd: MACDPoint
  bollinger: BollingerPoint
  sma_20: number | null
  sma_50: number | null
  sma_200: number | null
  ema_12: number | null
  ema_26: number | null
  atr: number | null
}

export interface TechnicalAnalysisResponse {
  symbol: string
  interval: string
  series: IndicatorPoint[]
  support_levels: number[]
  resistance_levels: number[]
}

export interface NewsItem {
  title: string
  source: string
  url: string
  published_at: string | null
  sentiment_score: number
  sentiment_label: 'positive' | 'neutral' | 'negative'
}

export interface FundamentalAnalysisResponse {
  symbol: string
  sentiment_score: number
  news_impact: number
  key_events: string[]
  news_items: NewsItem[]
  analyzed_at: string
}

File: backend/app/schemas/analysis.py
from datetime import datetime

from pydantic import BaseModel


class MACDPoint(BaseModel):
    macd: float | None
    signal: float | None
    histogram: float | None


class BollingerPoint(BaseModel):
    upper: float | None
    middle: float | None
    lower: float | None


class IndicatorPoint(BaseModel):
    timestamp: datetime
    close: float
    rsi: float | None
    macd: MACDPoint
    bollinger: BollingerPoint
    sma_20: float | None
    sma_50: float | None
    sma_200: float | None
    ema_12: float | None
    ema_26: float | None
    atr: float | None


class TechnicalAnalysisResponse(BaseModel):
    symbol: str
    interval: str
    series: list[IndicatorPoint]
    support_levels: list[float]
    resistance_levels: list[float]
